serializer/seata-serializer-seata/src/main/java/io/seata/serializer/seata/SeataSerializer.java

```diff
@@ -16,72 +16,43 @@
  */
 package io.seata.serializer.seata;
 
-import java.nio.ByteBuffer;
-
-import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
+import com.google.common.collect.ImmutableMap;
 import io.seata.common.loader.LoadLevel;
-import io.seata.common.util.BufferUtils;
-import io.seata.core.protocol.AbstractMessage;
+import io.seata.common.loader.Scope;
+import io.seata.core.protocol.ProtocolConstants;
 import io.seata.core.serializer.Serializer;
+import io.seata.serializer.seata.protocol.v0.SeataSerializerV0;
+import io.seata.serializer.seata.protocol.v1.SeataSerializerV1;
+
+import java.util.Map;
 
 /**
  * The Seata codec.
- *
  */
-@LoadLevel(name = "SEATA")
+@LoadLevel(name = "SEATA", scope = Scope.PROTOTYPE)
 public class SeataSerializer implements Serializer {
 
-    @Override
-    public <T> byte[] serialize(T t) {
-        if (!(t instanceof AbstractMessage)) {
-            throw new IllegalArgumentException("AbstractMessage isn't available.");
-        }
-        AbstractMessage abstractMessage = (AbstractMessage)t;
-        //typecode
-        short typecode = abstractMessage.getTypeCode();
-        //msg codec
-        MessageSeataCodec messageCodec = MessageCodecFactory.getMessageCodec(typecode);
-        //get empty ByteBuffer
-        ByteBuf out = Unpooled.buffer(1024);
-        //msg encode
-        messageCodec.encode(t, out);
-        byte[] body = new byte[out.readableBytes()];
-        out.readBytes(body);
+    Serializer versionSeataSerializer;
 
-        //typecode + body
-        ByteBuffer byteBuffer = ByteBuffer.allocate(2 + body.length);
-        byteBuffer.putShort(typecode);
-        byteBuffer.put(body);
+    static Map<Byte, Serializer> VERSION_MAP = ImmutableMap.<Byte, Serializer>builder()
+            .put(ProtocolConstants.VERSION_0, new SeataSerializerV0())
+            .put(ProtocolConstants.VERSION_1, new SeataSerializerV1())
+            .build();
 
-        BufferUtils.flip(byteBuffer);
-        byte[] content = new byte[byteBuffer.limit()];
-        byteBuffer.get(content);
-        return content;
+    public SeataSerializer(Byte version){
+        versionSeataSerializer =  VERSION_MAP.get(version);
```

Comment:
为什么需要一个map，我认为使用if else构建对应版本的序列化器即可，因为本身spi加载到时已经对SeataSerializer进行了缓存
Why do you need a map? I think it is enough to use if else to build the corresponding version of the serializer, because the SeataSerializer has already been cached when the spi is loaded.

Reply:
I will fix it